Sources/MapboxMaps/Foundation/BaseMapView.swift

```diff
@@ -483,6 +389,48 @@ open class BaseMapView: UIView, MapClient, MBMMetalViewProvider, CameraViewDeleg
     }
 }
 
+extension BaseMapView: DelegatingObserverDelegate {
+    /// Notify correct handler
+    internal func notify(for event: MapboxCoreMaps.Event) {
+        let handlers = eventHandlers[event.type]
+        handlers?.forEach { (handler) in
+            handler(event)
+        }
+    }
+}
```

Comment:
Can we move this to a new file called BaseMapView+DelefatingObserverDelegate

Reply:
We could, but it seems unnecessary to me. What problem would we be solving by moving it?

Reply:
I don't think it solves a problem, but if these extensions have the potential to grow in the future, it would be nice to have a bloated 500 + line map view file.

Reply:
If BaseMapView gets that big, the solution is probably to refactor it into multiple components rather than breaking it up into multiple files. I'd be in favor of leaving this as-is for now.